src/equations/1d/compressible_euler.jl

```diff
@@ -0,0 +1,838 @@
+  # Same settings as in `initial_conditions`
+  c = 2
+  A = 0.1
+  L = 2
+  f = 1/L
+  ω = 2 * pi * f
+  γ = equation.gamma
+
+  for i in 1:n_nodes
+    x1 = x[1, i, element_id]
+
+    si, co = sincos((t - x1)*ω)
+    tmp = (-((4 * si * A - 4c) + 1) * (γ - 1) * co * A * ω) / 2
+
+    ut[2, i, element_id] += tmp
+    ut[3, i, element_id] += tmp
+
+    # Original terms (without performanc enhancements)
+    # ut[1, i, element_id] += 0
+    # ut[2, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+    # ut[3, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+  end
+
+  return nothing
+end
+#=
+function source_terms_eoc_test_coupled_euler_gravity(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations2D)
+  # Same settings as in `initial_conditions_eoc_test_coupled_euler_gravity`
+  c = 2.0
+  A = 0.1
+  G = 1.0 # gravitational constant, must match coupling solver
+  C_grav = -2.0 * G / pi # 2 == 4 / ndims
+
+  for j in 1:n_nodes, i in 1:n_nodes
+    x1 = x[1, i, j, element_id]
+    x2 = x[2, i, j, element_id]
+    si, co = sincos(pi * (x1 + x2 - t))
+    rhox = A * pi * co
+    rho  = c + A *  si
+
+    ut[1, i, j, element_id] += rhox
+    ut[2, i, j, element_id] += rhox
+    ut[3, i, j, element_id] += rhox
+    ut[4, i, j, element_id] += (1.0 - C_grav*rho)*rhox
+  end
+
+  return nothing
+end
+
+function source_terms_eoc_test_euler(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations2D)
+  # Same settings as in `initial_conditions_eoc_test_coupled_euler_gravity`
+  c = 2.0
+  A = 0.1
+  G = 1.0
+  C_grav = -2 * G / pi # 2 == 4 / ndims
+
+  for j in 1:n_nodes, i in 1:n_nodes
+    x1 = x[1, i, j, element_id]
+    x2 = x[2, i, j, element_id]
+    si, co = sincos(pi * (x1 + x2 - t))
+    rhox = A * pi * co
+    rho  = c + A *  si
+
+    ut[1, i, j, element_id] += rhox
+    ut[2, i, j, element_id] += rhox * (1 -     C_grav * rho)
+    ut[3, i, j, element_id] += rhox * (1 -     C_grav * rho)
+    ut[4, i, j, element_id] += rhox * (1 - 3 * C_grav * rho)
+  end
+
+  return nothing
+end
+=#
+# Empty source terms required for coupled Euler-gravity simulations
+function source_terms_harmonic(ut, u, x, element_id, t, n_nodes, equation::CompressibleEulerEquations1D)
+  # OBS! used for the Jeans instability as well as self-gravitating Sedov blast
+  # TODO: make this cleaner and let each solver have a different source term name
+  return nothing
+end
+
+# Calculate 1D flux for a single point
+@inline function calcflux(u, orientation, equation::CompressibleEulerEquations1D)
+  rho, rho_v1, rho_e = u
+  v1 = rho_v1/rho
+  p = (equation.gamma - 1) * (rho_e - 1/2 * rho * v1^2)
+  if orientation == 1
+    f1 = rho_v1
+    f2 = rho_v1 * v1 + p
+    f3 = (rho_e + p) * v1
+  end
```

Comment:
```suggestion
  # Ignore orientation since it is always "1" in 1D
  f1 = rho_v1
  f2 = rho_v1 * v1 + p
  f3 = (rho_e + p) * v1
```
No unnecessary checks, as they just slow us down